aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -526,11 +527,8 @@ std::optional<Tensor> convert_boolean_attn_mask(const std::optional<Tensor>& att
   // Convert boolean mask to additive mask; need to invert mask to indicate what
   // to mask *out*.
   if (attn_mask->dtype() == at::kBool) {
-    auto new_attn_mask = at::zeros_like(attn_mask.value(), dtype);
     // TODO Use the max type of the input and output
-    new_attn_mask.masked_fill_(
-        attn_mask->logical_not(), -std::numeric_limits<double>::infinity());
-    return new_attn_mask;
+    return at::where(attn_mask->logical_not(), -std::numeric_limits<double>::infinity(), at::scalar_tensor(0.0, at::TensorOptions().dtype(dtype)));
```

Comment:
NIT: You can also avoid the `logical_not` by reversing the argument order.
```suggestion
    return at::where(*attn_mask, 0.0, at::scalar_tensor(-std::numeric_limits<double>::infinity(), at::TensorOptions().dtype(dtype)));
```